controller/src/beerocks/master/wfa_ca.cpp

```diff
@@ -73,6 +73,106 @@ static bool validate_hex_notation(const std::string &str, uint8_t expected_octet
+ * @param[out] bss_info_conf Controller database struct filled with configuration.
+ * @param[out] err_string Contains an error description if the function fails.
+ * @return true if successful, false if not.
+ */
+static std::string parse_bss_info(const std::string &bss_info_str,
+                                  db::bss_info_conf_t &bss_info_conf, std::string &err_string)
+{
+    auto confs = string_utils::str_split(bss_info_str, ' ');
+    if (confs.size() != 8) {
+        err_string = "missing configuration";
+        return std::string();
+    }
+
+    // Alid
+    std::string al_mac(confs[0]);
+    if (!net::network_utils::is_valid_mac(al_mac)) {
+        err_string = "invalid al_mac";
+        return std::string();
+    }
+
+    bss_info_conf = {};
+
+    // Operating class
+    uint operating_class = string_utils::stoi(confs[1]);
+    if (operating_class > UINT8_MAX || operating_class == 0) {
+        err_string = "invalid operating class";
+        return std::string();
+    }
+    bss_info_conf.operating_class = operating_class;
+
+    // SSID
+    bss_info_conf.ssid = confs[2];
+    if (bss_info_conf.ssid.size() > WSC::eWscLengths::WSC_MAX_SSID_LENGTH) {
+        err_string = "ssid is too long";
+        return std::string();
+    }
+
+    // Authentication type
+    auto &authentication_type_str = confs[3];
+    if (!validate_hex_notation(authentication_type_str, 2)) {
+        err_string = "invalid authentication type format";
+        return std::string();
+    }
+
+    uint16_t authentication_type = std::stoi(authentication_type_str, nullptr, 16);
+
+    if (!WSC::eWscAuthValidate::check(authentication_type)) {
+        err_string = "invalid authentication type value";
+        return std::string();
+    }
+    bss_info_conf.authentication_type = static_cast<WSC::eWscAuth>(authentication_type);
+
+    // Encryption type
+    auto &encryption_type_str = confs[4];
+    if (!validate_hex_notation(encryption_type_str, 2)) {
+        err_string = "invalid encryption type format";
+        return std::string();
+    }
+
+    uint16_t encryption_type = std::stoi(encryption_type_str, nullptr, 16);
+
+    if (!WSC::eWscEncrValidate::check(encryption_type)) {
+        err_string = "invalid encryption type value";
+        return std::string();
+    }
+    bss_info_conf.encryption_type = static_cast<WSC::eWscEncr>(encryption_type);
+
+    // Network key
+    bss_info_conf.network_key = confs[5];
+    if (bss_info_conf.network_key.size() > WSC::eWscLengths::WSC_MAX_NETWORK_KEY_LENGTH) {
+        err_string = "network key is too long";
+        return std::string();
+    }
+
+    // Bit 6 of Multi-AP IE's extention attribute, aka "Backhaul BSS"
+    auto &bit_6_str = confs[6];
+    if (bit_6_str != "0" && bit_6_str != "1") {
+        err_string = "invalid bit 6 of Multi-AP IE's extention attribute";
+        return std::string();
+    }
+    uint8_t bss_type = (bit_6_str == "1" ? WSC::eWscVendorExtSubelementBssType::BACKHAUL_BSS : 0);
```

Comment:
For symmetry, I think it's nicer to set `bss_info_conf.bss_type = 0;` before, and then use `bss_info_conf.bss_type |= ...` both here and for the fronthaul.

Argh, that's not possible because eWscVendorExtSubelementBssType doesn't have an `operator|` :-)

Reply:
been there 😆